packages/runtime-common/matrix-client.ts

```diff
@@ -43,6 +66,20 @@ export class MatrixClient {
   }
 
   async login() {
+    let password: string | undefined;
+    if (this.password) {
+      password = this.password;
+    } else if (this.seed) {
+      let hash = new Sha256();
+      hash.update(this.username);
+      hash.update(this.seed);
+      password = uint8ArrayToHex(await hash.digest());
```

Comment:
I opted to perform the password calculation from the seed in the matrix client because it is async (as opposed to in the Realm constructor where we create the client)